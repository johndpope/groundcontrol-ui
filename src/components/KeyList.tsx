// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import graphql from "babel-plugin-relay/macro";
import React from "react";
import { createFragmentContainer } from "react-relay";
import { List } from "semantic-ui-react";

import { KeyList_items } from "./__generated__/KeyList_items.graphql";

import KeyListItem, { IProps as IKeyListItemProps } from "./KeyListItem";

export interface IProps {
  items: KeyList_items;
  onEdit: (values: IKeyListItemProps) => any;
  onDelete: (values: IKeyListItemProps) => any;
}

export function KeyList({ items, onEdit, onDelete }: IProps) {
  if (items.length < 1) {
    return <p>There are no keys at this time.</p>;
  }

  const listItems = items.map(item => (
    <KeyListItem
      key={item.id}
      item={item}
      onEdit={onEdit}
      onDelete={onDelete}
    />
  ));

  return <List divided={true}>{listItems}</List>;
}

export default createFragmentContainer(
  KeyList,
  graphql`
    fragment KeyList_items on Key @relay(plural: true) {
      id
      name
      value
      ...KeyListItem_item
    }
  `
);
