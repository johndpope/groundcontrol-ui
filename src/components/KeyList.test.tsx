// Copyright 2019 Stratumn
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { shallow } from "enzyme";
import React from "react";

import { mockQueryPropAttrs } from "../testing/relay";

import { KeyList } from "./KeyList";

const props = {
  items: [
    {
      ...mockQueryPropAttrs(),
      id: "id1",
      name: "name1",
      value: "value1"
    },
    {
      ...mockQueryPropAttrs(),
      id: "id2",
      name: "name2",
      value: "value2"
    }
  ],
  onDelete: jest.fn(),
  onEdit: jest.fn()
};

describe("<KeyList />", () => {
  it("renders items correctly", () => {
    const wrapper = shallow(<KeyList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly without items", () => {
    const wrapper = shallow(<KeyList {...props} items={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
