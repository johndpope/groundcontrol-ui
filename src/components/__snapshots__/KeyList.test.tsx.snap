// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KeyList /> renders correctly without items 1`] = `
<p>
  There are no keys at this time.
</p>
`;

exports[`<KeyList /> renders items correctly 1`] = `
<List
  divided={true}
>
  <ForwardRef(Relay(KeyListItem))
    item={
      Object {
        " $fragmentRefs": null,
        " $refType": null,
        "id": "id1",
        "name": "name1",
        "value": "value1",
      }
    }
    key="id1"
    onDelete={[MockFunction]}
    onEdit={[MockFunction]}
  />
  <ForwardRef(Relay(KeyListItem))
    item={
      Object {
        " $fragmentRefs": null,
        " $refType": null,
        "id": "id2",
        "name": "name2",
        "value": "value2",
      }
    }
    key="id2"
    onDelete={[MockFunction]}
    onEdit={[MockFunction]}
  />
</List>
`;
